core/parentchain/block-import-dispatcher/src/retain_latest_dispatcher.rs

```diff
@@ -0,0 +1,147 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! A block import dispatcher that retains the latest block until its import is explicitly triggered.
+
+use crate::{
+	error::{Error, Result},
+	DispatchBlockImport,
+};
+use itc_parentchain_block_import_queue::{PopFromBlockQueue, PushToBlockQueue};
+use itc_parentchain_block_importer::ImportParentchainBlocks;
+use std::{sync::Arc, vec::Vec};
+
+/// Trait to specifically trigger the import of the latest retained parentchain block.
+pub trait TriggerLatestParentchainBlockImport {
+	fn import_latest(&self) -> Result<()>;
+}
+
+/// Dispatcher for block imports that always retains the latest parentchain block.
+/// Importing the latest block has to be triggered specifically.
+///
+/// Uses a queue internally. Upon receiving a dispatch request, it imports all blocks in the queue
+/// except the latest one.
+pub struct RetainLatestDispatcher<BlockImporter, BlockImportQueue> {
+	block_importer: Arc<BlockImporter>,
+	block_import_queue: Arc<BlockImportQueue>,
+}
```

Comment:
This dispatcher implementation, as the name suggests, will retain the latest parentchain block. Any time it receives a dispatch request, it will import everyting except the latest one. Importing the latest block can only be triggered using a separate trait implementation of `TriggerLatestParentchainBlockImport`. This will be used by the sidechain to sync with its block production (-> next PR)